### PR TITLE
[SPARK-38081][K8S][TESTS] Support `cloud`-backend in K8s IT with SBT

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to
- Support `cloud` backend in K8s IT with SBT (Image building/pushing/testing)
- Add a new K8s test tag, `local`, and apply it to a test case using local HTTP server.

### Why are the changes needed?

To run K8s IT in the cloud environment more easily.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Manually test like the following.
```
$ build/sbt -Psparkr -Pkubernetes -Pkubernetes-integration-tests -Dtest.exclude.tags=minikube,local -Dspark.kubernetes.test.deployMode=cloud -Dspark.kubernetes.test.master=k8s://....eks.amazonaws.com -Dspark.kubernetes.test.namespace=spark-cloud-test -Dspark.kubernetes.test.imageRepo=... -Dspark.kubernetes.test.imageTag=2022-02-01 "kubernetes-integration-tests/test"

...
[info] KubernetesSuite:
[info] - Run SparkPi with no resources (26 seconds, 678 milliseconds)
[info] - Run SparkPi with no resources & statefulset allocation (18 seconds, 617 milliseconds)
[info] - Run SparkPi with a very long application name. (17 seconds, 205 milliseconds)
[info] - Use SparkLauncher.NO_RESOURCE (17 seconds, 555 milliseconds)
[info] - Run SparkPi with a master URL without a scheme. (17 seconds, 478 milliseconds)
[info] - Run SparkPi with an argument. (17 seconds, 518 milliseconds)
[info] - Run SparkPi with custom labels, annotations, and environment variables. (17 seconds, 648 milliseconds)
[info] - All pods have the same service account by default (17 seconds, 800 milliseconds)
[info] - Run extraJVMOptions check on driver (11 seconds, 141 milliseconds)
[info] - Verify logging configuration is picked from the provided SPARK_CONF_DIR/log4j2.properties (25 seconds, 608 milliseconds)
[info] - Run SparkPi with env and mount secrets. (27 seconds, 114 milliseconds)
[info] - Run PySpark on simple pi.py example (42 seconds, 929 milliseconds)
[info] - Run PySpark to test a pyfiles example (19 seconds, 914 milliseconds)
[info] - Run PySpark with memory customization (16 seconds, 985 milliseconds)
[info] - Run in client mode. (10 seconds, 42 milliseconds)
[info] - Start pod creation from template (16 seconds, 207 milliseconds)
[info] - Test basic decommissioning (49 seconds, 519 milliseconds)
[info] - Test basic decommissioning with shuffle cleanup (49 seconds, 472 milliseconds)
[info] - Test decommissioning with dynamic allocation & shuffle cleanups (2 minutes, 49 seconds)
[info] - Test decommissioning timeouts (50 seconds, 423 milliseconds)
[info] - SPARK-37576: Rolling decommissioning (1 minute, 13 seconds)
[info] - Run SparkR on simple dataframe.R example (51 seconds, 712 milliseconds)
[info] Run completed in 15 minutes, 50 seconds.
[info] Total number of tests run: 22
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 22, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 1920 s (32:00), completed Jan 31, 2022 11:56:38 PM
```

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @williamhyun

**Comment by dongjoon-hyun** [comment]
Could you review this PR, @viirya ?

File: resource-managers/kubernetes/integration-tests/README.md [hunk 1]
```diff
@@ -187,7 +187,7 @@ to the wrapper scripts and using the wrapper scripts will simply set these appro
   <tr>
     <td><code>spark.kubernetes.test.master</code></td>
     <td>
-      When using the <code>cloud-url</code> backend must be specified to indicate the K8S master URL to communicate
```
Comment (viirya) [inline resource-managers/kubernetes/integration-tests/README.md:190]: Is there a `cloud-url` backend before?

**Review by viirya — APPROVED** [review]
The document `integration-tests/README.md` seems not mention how to run K8S IT using sbt. Should we update it too?

File: resource-managers/kubernetes/integration-tests/README.md (hunk 1 above)
Comment (dongjoon-hyun) [reply resource-managers/kubernetes/integration-tests/README.md:190]: No~ This was a typo. We have only `cloud` backend.

**Comment by dongjoon-hyun** [comment]
Yes, there are several improvements in SBT area. Since it's broad, let me make a separate doc PR for that, @viirya .
> The document integration-tests/README.md seems not mention how to run K8S IT using sbt. Should we update it too?

```
(This PR)      [SPARK-38081][K8S][TESTS] Support cloud-backend in K8s IT with SBT
419d17378c [SPARK-38072][K8S][TESTS] Support K8s imageTag parameter in SBT K8s IT
71c34b42f3 [SPARK-38071][K8S][TESTS] Support K8s namespace parameter in SBT K8s IT
ed3ea989f9 [SPARK-37713][K8S][FOLLOWUP] Fix insufficient namespace propagation
169be3e1e6 [SPARK-38049][K8S][TESTS] Use Java 17 in K8s integration tests by default
4f75577234 [SPARK-38048][K8S][TESTS] Add `IntegrationTestBackend.describePods` to support all K8s test backends
69c213d356 [SPARK-38029][K8S][TESTS] Support K8S integration test in SBT
```

Merged to master~

File: resource-managers/kubernetes/integration-tests/README.md (hunk 1 above)
Comment (viirya) [reply resource-managers/kubernetes/integration-tests/README.md:190]: Okay, got it. Thanks.

**Comment by dongjoon-hyun** [comment]
I backported this test-only patch to branch-3.2 for Apache Spark 3.2.2. This is verified with `EKS Graviton (ARM64)` cluster directly on the latest `branch-3.2` + this patch.
```
$ build/sbt -Psparkr -Pkubernetes -Pkubernetes-integration-tests -Dtest.exclude.tags=minikube,local -Dspark.kubernetes.test.deployMode=cloud -Dspark.kubernetes.test.master=k8s://https://....eks.amazonaws.com -Dspark.kubernetes.test.imageRepo=... -Dspark.kubernetes.test.imageTag=20220302 "kubernetes-integration-tests/test"
...
[info] KubernetesSuite:
[info] - Run SparkPi with no resources (55 seconds, 72 milliseconds)
[info] - Run SparkPi with a very long application name. (14 seconds, 66 milliseconds)
[info] - Use SparkLauncher.NO_RESOURCE (13 seconds, 491 milliseconds)
[info] - Run SparkPi with a master URL without a scheme. (13 seconds, 515 milliseconds)
[info] - Run SparkPi with an argument. (13 seconds, 599 milliseconds)
[info] - Run SparkPi with custom labels, annotations, and environment variables. (13 seconds, 895 milliseconds)
[info] - All pods have the same service account by default (14 seconds, 12 milliseconds)
[info] - Run extraJVMOptions check on driver (7 seconds, 538 milliseconds)
[info] - Verify logging configuration is picked from the provided SPARK_CONF_DIR/log4j.properties (15 seconds, 350 milliseconds)
[info] - Run SparkPi with env and mount secrets. (23 seconds, 595 milliseconds)
[info] - Run PySpark on simple pi.py example (33 seconds, 942 milliseconds)
[info] - Run PySpark to test a pyfiles example (18 seconds, 532 milliseconds)
[info] - Run PySpark with memory customization (15 seconds, 747 milliseconds)
[info] - Run in client mode. (10 seconds, 351 milliseconds)
[info] - Start pod creation from template (13 seconds, 869 milliseconds)
[info] - Test basic decommissioning (47 seconds, 300 milliseconds)
[info] - Test basic decommissioning with shuffle cleanup (48 seconds, 170 milliseconds)
[info] - Test decommissioning with dynamic allocation & shuffle cleanups (2 minutes, 49 seconds)
[info] - Test decommissioning timeouts (47 seconds, 304 milliseconds)
[info] - Run SparkR on simple dataframe.R example (54 seconds, 275 milliseconds)
[info] Run completed in 13 minutes, 29 seconds.
[info] Total number of tests run: 20
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 20, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[success] Total time: 1059 s (17:39), completed Mar 2, 2022, 3:01:10 PM
```

**Pull request closed without merging**